pkgs/by-name/fl/flatpak/package.nix

```diff
@@ -89,163 +108,145 @@ stdenv.mkDerivation (finalAttrs: {
 
-    # The icon validator needs to access the gdk-pixbuf loaders in the Nix store
-    # and cannot bind FHS paths since those are not available on NixOS.
-    finalAttrs.passthru.icon-validator-patch
-
     # Try mounting fonts and icons from NixOS locations if FHS locations don't exist.
     # https://github.com/NixOS/nixpkgs/issues/119433
     ./fix-fonts-icons.patch
 
-    # TODO: Remove when updating to 1.16
-    # Ensure flatpak uses the system's zoneinfo from $TZDIR
-    # https://github.com/NixOS/nixpkgs/issues/238386
-    (fetchpatch {
-      url = "https://github.com/flatpak/flatpak/pull/5850/commits/a8a35bf4d9fc3d76e1a5049a6a591faec04a42fd.patch";
-      hash = "sha256-JqkPbnzgZNZq/mplZqohhHFdjRrvYFjE4C02pI3feBo=";
-    })
-    (fetchpatch {
-      url = "https://github.com/flatpak/flatpak/pull/5850/commits/5ea13b09612215559081c27b60df4fb720cb08d5.patch";
-      hash = "sha256-BWbyQ2en3RtN4Ec5n62CULAhvywlQLhcl3Fmd4fsR1s=";
-    })
-    (fetchpatch {
-      url = "https://github.com/flatpak/flatpak/pull/5850/commits/7c8a81f08908019bbf69358de199748a9bcb29e3.patch";
-      hash = "sha256-RiG2jPmG+Igskxv8oQquOUYsG4srgdMXWe34ojMXslo=";
-    })
-  ];
+    # Nix environment hacks should not leak into the apps.
+    # https://github.com/NixOS/nixpkgs/issues/53441
+    ./unset-env-vars.patch
 
-  nativeBuildInputs = [
-    autoreconfHook
-    libxml2
-    docbook_xml_dtd_45
-    docbook-xsl-nons
-    which
-    gobject-introspection
-    gtk-doc
-    intltool
-    libxslt
-    pkg-config
-    xmlto
-    bison
-    wrapGAppsNoGuiHook
+    # The icon validator needs to access the gdk-pixbuf loaders in the Nix store
+    # and cannot bind FHS paths since those are not available on NixOS.
+    finalAttrs.passthru.icon-validator-patch
   ];
 
-  buildInputs = [
-    appstream
-    bubblewrap
-    bzip2
-    curl
-    dbus
-    dconf
-    gpgme
-    json-glib
-    libarchive
-    libcap
-    libseccomp
-    libxml2
-    xz
-    zstd
-    polkit
-    python3
-    systemd
-    xorg.libXau
-    fuse3
-    gsettings-desktop-schemas
-    glib-networking
-    librsvg # for flatpak-validate-icon
-  ] ++ lib.optionals withGtkDoc [ gtk-doc ];
+  # Fixup shebangs in some scripts
+  #
+  # Don't prefix the already absolute `man` directory with the install prefix
+  postPatch = ''
+    patchShebangs buildutil/ tests/
+    patchShebangs --build subprojects/variant-schema-compiler/variant-schema-compiler
+
+    substituteInPlace doc/meson.build \
+      --replace-fail '$MESON_INSTALL_DESTDIR_PREFIX/@1@/@2@' '@1@/@2@'
+  '';
+
+  strictDeps = true;
+
+  nativeBuildInputs =
+    [
+      (python3.pythonOnBuildForHost.withPackages (p: [ p.pyparsing ]))
+      bison
+      gobject-introspection
+      meson
+      ninja
+      pkg-config
+      validatePkgConfig
+      wrapGAppsNoGuiHook
+    ]
+    ++ lib.optional withGtkDoc gtk-doc
+    ++ lib.optional withMan libxslt
+    ++ lib.optional withSELinuxModule bzip2
+    ++ lib.optionals withDocbookDocs [
```

Comment:
How about we combine the docs into one doc setting?

Reply:
We could *maybe* merge the docbook and man outputs, but the gtkdoc stuff requires native binaries to run; I'd rather not block all doc generation behind it. The docbooks and manpages are also different enough that I think distinct options would be a good way to go